controllers/configurationpolicy_controller.go

```diff
@@ -2382,13 +2415,28 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 
 		if keyUpdateNeeded {
 			if strings.EqualFold(string(remediation), string(policyv1.Inform)) {
-				return true, "", false
+				return true, "", false, false
 			} else if isStatus {
 				statusUpdated = true
 				log.Info("Ignoring an update to the object status", "key", key)
 			} else {
 				updateNeeded = true
 				log.Info("Queuing an update for the object due to a value mismatch", "key", key)
+
+				updateParent := createStatus("", obj.gvr.Resource, map[string]map[string]interface{}{
```

Comment:
Could you please move this code to the calling method (handleSingleObj)? That way this method is scoped to just updating the object if needed and the the calling method handles all the status updates.

This makes the responsibilities tighter and clearer for each method.